File: frontend/src/types/topic.ts
export type TopicStatus = 'spike' | 'rising' | 'stable';

export interface Genre {
  id: string;
  slug: string;
  label: string;
  description: string;
  sortOrder: number;
}

export interface TopicSparklineData {
  timestamp: string;
  value: number;
}

export interface SpikeHistoryEntry {
  id: string;
  timestamp: string;
  peakZScore: number;
  status: TopicStatus;
  summary: string;
  durationMinutes: number;
}

export interface PostingTips {
  peakDays: string[];
  peakHoursStart: number;
  peakHoursEnd: number;
  nextSuggestedTime: string;
}

export interface Topic {
  id: string;
  name: string;
  keywords: string[];
  genre: string;
  status: TopicStatus;
  changePercent: number;
  zScore: number | null;
  currentVolume: number;
  baselineVolume: number;
  sparklineData: TopicSparklineData[];
  context: string | null;
  contextSummary: string | null;
  spikeStartedAt: string | null;
  weeklySparklineData: TopicSparklineData[];
  spikeHistory: SpikeHistoryEntry[];
  postingTips: PostingTips | null;
  notificationEnabled: boolean;
  createdAt: string;
}

File: backend/internal/domain/entity/topic.go
package entity

import "time"

// TopicStatus represents the current trend status of a topic.
type TopicStatus int

const (
	TopicSpike  TopicStatus = 1
	TopicRising TopicStatus = 2
	TopicStable TopicStatus = 3
)

// Topic represents a monitored trend topic.
// Genre は genres マスターテーブルの slug で、GenreID は FK。
// GenreSlug は JOIN/Preload で取得され、API レスポンスの genre フィールドに設定される。
type Topic struct {
	ID                  string
	Name                string
	Keywords            []string
	GenreID             string
	GenreSlug           string // genres テーブルから JOIN で取得
	Status              TopicStatus
	ChangePercent       float64
	ZScore              *float64
	CurrentVolume       int32
	BaselineVolume      int32
	Context             *string
	ContextSummary      *string
	SpikeStartedAt      *time.Time
	NotificationEnabled bool
	CreatedAt           time.Time
	UpdatedAt           time.Time
}
